### PR TITLE
Send less MessageLogger calls when doing job summary

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

The messages are now sent in batches of 10.

#### PR validation:

Code compiles. Framework unit tests pass. A quick visual inspection of a test job seems to generate expected output.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Framework/src/Schedule.cc [hunk 1]
```diff
@@ -785,6 +785,23 @@ namespace edm {
...
       return;
+
+    //Function to loop over items in a container and periodically
+    // flush to the message logger.
+    auto logForEach = [](auto const& iContainer, auto iMessage) {
+      auto logger = LogFwkVerbatim("FwkSummary");
+      int count = 0;
+      for (auto const& element : iContainer) {
+        iMessage(logger, element);
+        if (++count == 10) {
```
Comment (makortel) [inline FWCore/Framework/src/Schedule.cc:796]: Why not appending all the elements of `iContainer` to the `logger`? To avoid constructing a huge string?

Comment (Dr15Jones) [reply FWCore/Framework/src/Schedule.cc:796]: Yes. I was concerned we could create a several MB sized string for the HLT cases. Given we have seen memory spikes at the end of the job, I didn't want to accidentally add to that.

**Comment by makortel** [comment]
+core

**Comment by perrotta** [comment]
+1

**Pull request merged**